### PR TITLE
Added a domain blacklist parameter for the Email validator

### PR DESCRIPTION
**Pull request by GevatterGaul**
Hi, we added a domain blacklist parameter for the Email validator. The validation will fail for Emails from a domain on the blacklist with a message like this: `The domain of the email address is blacklisted (the portion after the @: foo.bar.com)`

We also added documentation and tests for this feature, as well as german and spanish translations for the error message.

### CODE REVIEW

File: formencode/validators.py [hunk 1]
```diff
@@ -1351,6 +1364,10 @@ def _validate_python(self, value, state):
             raise Invalid(
                 self.message('badUsername', state, username=username),
                 value, state)
+        if hasattr(self, 'domain_blacklist') and domain in self.domain_blacklist:
```
Comment (lambacck) [inline formencode/validators.py]: Domain names are case insensitive but there is no handling of that here. You could have ```mybaddomainname.com``` in ```domain_blacklist``` and the user could get by this check with ```haha@MyBadDomainName.com```. There should probably be some case normalization of ```domain_blacklist``` also.

**Comment by lambacck** [comment]
Thanks for the pull request. I think this is a good idea and I'm willing to merge once the case sensitivity issue is worked out.

**Comment by GevatterGaul** [comment]
Your are right with the case sensitivity. I will fix this in the near future.

**Comment by lambacck** [comment]
In order to merge this I need you to add yourself to the CONTRIBUTORS.txt file.

**Comment by lambacck** [comment]
Closing due to inactivity.

**Pull request closed without merging**